File: 09-05-2024_Maximize_Happiness_of_Selected_Children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        max_sum = 0
        for i in range(k):
            happiness[i] -= i
            if happiness[i] > 0:
                max_sum += happiness[i]
            else:
                break
        return max_sum

File: 09-05-2024_Maximize_Happiness_of_Selected_Children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
         sort(happiness.begin(),happiness.end(),greater<int>());
        long long maxSum=0;
        for(int i=0;i<k;i++){
            happiness[i]-=i;
            if(happiness[i]>0)
              maxSum+=happiness[i];
            else
              break;  
        }
        return maxSum;
    }
};
